### DISCUSSION TITLE
Adding a variable not associated to FE

### DISCUSSION THREAD

**Discussion by jfbarthelemy** [Ferrite-FEM/Ferrite.jl#814] [Q&A] [answered]
Dear Ferrite devs and users,

Would it be possible to add a global scalar (not a field attached to any interpolation system) as dof in a problem? I explain a case where I would need this. Consider a stready-state heat equation problem (possibly with non linear conductivity to avoid invoking linearity as a workaround later) on a domain such that one part Γ of the boundary is submitted to an unknown uniform temperature but the resulting heat flux (integral over Γ) is known. This problem is well posed and could for example be handled by introducing a new variable which is the unknown temperature and a Lagrange multiplier field on Γ to ensure the condition of uniformity of T on Γ. Is this feasible with Ferrite?

Thanks in advance

**Comment by KnutAM** [comment]
Some initial work here: https://github.com/Ferrite-FEM/Ferrite.jl/pull/596

Should really be included and isn't that hard. The main reason for not being included as of now is, IMO, the need to define a general interface how this should be handled as it can be used for a range of different things. Feel free to comment on that PR as well (even if outdated)!

**Reply by jfbarthelemy** [reply]
Thanks

**Comment by fredrikekre** [comment, marked as answer]
For now I think you have to handle this on your own by constructing the individual blocks, and then concatenating, e.g.
```
A  B
B' C
```
where `A` (ndofs x ndofs) is the sparse matrix from Ferrite, `B` (ndofs x 1) a matrix you put together on your own, and `C` (1 x 1) just a zero matrix, I would assume.

**Reply by jfbarthelemy** [reply]
Thanks. I'll try to do that.
